File: backend/Services/Parsers/AbsurdleParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class AbsurdleParser : BasicScoreResultParser
    {
        private ILogger<AbsurdleParser> _logger;

        public AbsurdleParser(ILogger<AbsurdleParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => false;
        public override string GameName => "Absurdle";
        public override string? HelpText => "Start with today's random word.";
        protected override Regex Parser => new Regex($"{GameName} (?<{ScoreGroup}>\\d+)/");
        protected override string? ExtraContent => Url;
        public override string Url => "https://qntm.org/files/absurdle/absurdle.html";
    }
}

File: lambdas/src/parsers/absurdle-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class AbsurdleParser extends BasicScoreResultParser {
  readonly countWinner = false;
  readonly gameName = 'Absurdle';
  readonly helpText = 'Start with today\'s random word.';
  readonly url = 'https://qntm.org/files/absurdle/absurdle.html';
  protected readonly extraContent = this.url;
  
  protected readonly parser = new RegExp(
    `${this.gameName} (?<${BasicScoreResultParser.SCORE_GROUP}>\\d+)/`
  );
}
